### PR TITLE
feat: add node_manager implementation

### PR DESCRIPTION
**Pull request by EstebanBAR0N**
In this pull request, we can find all functions to get node-agent resources (Memory, CPU and Disk - limit/usage).

In addition, unit tests are implemented.

For the functions of the disks, I made a choice. The choice is to consider the biggest disk (in GB) as the main disk.

### CODE REVIEW

File: node-agent/node_manager/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,165 @@
...
+const KB_TO_MB: u64 = 1000;
+const BIT_TO_GB: u64 = 1000000000;
+
+// used in all functions to access system
+fn system() -> System {
+    let mut sys = System::new_all();
+    sys.refresh_all();
+
+    sys
+}
```
Comment (sameo) [inline node-agent/node_manager/src/lib.rs:14]: We're recreating a new `System` instance for every `system()` call, while the only thing we should be doing is refreshing it.

Please consider creating a static instance of `System` with e.g. the [lazy_static](https://crates.io/crates/lazy_static) crate instead and only refresh it when needed.

Comment (sameo) [reply node-agent/node_manager/src/lib.rs:14]: Another option would be to have a `NodeSystem` structure:

```Rust
pub struct NodeSystem {
    inner: System,
}
```

define methods on it, and have your node manager instance carry one `NodeSystem` instance. It's probably a more elegant solution than a static `System` instance.

File: node-agent/node_manager/Cargo.toml [hunk 2]
```diff
@@ -5,4 +5,9 @@ edition = "2021"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
+[lib]
+name = "node_manager"
+path = "src/lib.rs"
+
```
Comment (sameo) [inline node-agent/node_manager/Cargo.toml:11]: I think that's probably not needed.

File: node-agent/node_manager/src/lib.rs [hunk 3]
```diff
@@ -1,3 +1,165 @@
...
+// ------ MEMORY ------
+
+/*
+  Returns the limit memory space (in MB)
+*/
+pub fn total_memory() -> u64 {
+    let sys: System = system();
+
+    let total_memory: u64 = (sys.available_memory() + sys.used_memory()) / KB_TO_MB;
+    println!("total memory: {:?} MB", &total_memory);
```
[the remark this answers is not in the document]
Comment (EstebanBAR0N) [reply node-agent/node_manager/src/lib.rs:62]: Ok, I hadn't planned on that, thanks for your feedback.

I will change that.

File: node-agent/node_manager/src/lib.rs (hunk 1 above)
Comment (EstebanBAR0N) [reply node-agent/node_manager/src/lib.rs:14]: You are right, I think I will change that with your second option.

Thank you.

File: node-agent/node_manager/Cargo.toml (hunk 2 above)
Comment (EstebanBAR0N) [reply node-agent/node_manager/Cargo.toml:11]: Yes, you are rigth, It's old stuff, I will remove that.

Thank you.

**Review by sameo — APPROVED** [review]

**Comment by sameo** [comment]
Please squash the 2 commits into one.

**Comment by EstebanBAR0N** [comment]
The two previous commits were squashed

**Review by sameo — APPROVED** [review]

**Pull request merged**